common/persistence/sql/storage/sqldb/interfaces.go

```diff
@@ -205,14 +205,29 @@ type (
 		DataEncoding string
 	}
 
-	// ReplicationTasksFilter contains the column names within domain table that
+	// ReplicationTaskDLQRow represents a row in replication_tasks_dlq table
+	ReplicationTaskDLQRow struct {
+		SourceClusterName string
+		ShardID           int
+		TaskID            int64
+		Data              []byte
+		DataEncoding      string
+	}
+
+	// ReplicationTasksFilter contains the column names within replication_tasks table that
 	// can be used to filter results through a WHERE clause
 	ReplicationTasksFilter struct {
 		ShardID   int
-		TaskID    *int64
-		MinTaskID *int64
-		MaxTaskID *int64
-		PageSize  *int
+		MinTaskID int64
+		MaxTaskID int64
+		PageSize  int
```

Comment:
any reason to changing from pointer?
seems sql side convention is using pointer

Reply:
We don't need them to be pointers at all. We used to use pointer and then dereference it in SQL layer.